python/ray/worker.py

```diff
@@ -1516,22 +1519,43 @@ def get(object_refs, *, timeout=None):
 
 
 @client_mode_hook
-def put(value):
+def put(value, *, _owner=None):
     """Store an object in the object store.
 
     The object may not be evicted while a reference to the returned ID exists.
 
     Args:
         value: The Python object to be stored.
+        _owner: The `ActorHandle` or serialized address of object's owner.
 
     Returns:
         The object ref assigned to this value.
     """
     worker = global_worker
     worker.check_connected()
+
+    if _owner is None:
+        serialize_owner_address = None
+    elif isinstance(_owner, bytes):
+        serialize_owner_address = _owner
+    elif isinstance(_owner, ray.actor.ActorHandle):
+        # Ensure `ray.state.state.global_state_accessor` is not None
+        ray.state.state._check_connected()
+        owner_address = ray.gcs_utils.ActorTableData.FromString(
```

Comment:
Can we do this in core worker instead of in Python?

Reply:
I think what you said is reasonable, I will change it later

Reply:
I tried it, but I didn’t find a good way to achieve it.

If we do this in c++ CoreWorker, it makes `CoreWorker::Put` become very strange.
If we do this in cython(_raylet.pyx), this requires us to initialize and manage cython class `GlobalStateAccessor`, which seems to be a bit more to be done.
If you don't like this implement, I can implement it on `ray.state.GlobalState`, just like `ray.state.GlobalState.actor_table`